README.md

```diff
@@ -1,23 +1,78 @@
+(to change the default MTU add `--opt com.docker.network.driver.mtu=MTU_VALUE`)
 
-    $ docker service ls
-    ID                  NAME                     MODE                REPLICAS            IMAGE                            PORTS
-    813gplgsyi6z        mongo_mongo              global              3/3                 mongo:3.2                        *:27017->27017/tcp
-    g5rxax0gucbn        mongo_mongo-controller   replicated          1/1                 martel/mongo-replica-ctrl:test   
+Then, simply run
 
-The mongo service need to be deployed in "global" mode if you want to use storage persistence (this is to avoid that more than one instance is deployed on the same node).
+* `docker stack deploy -c docker-compose.yml STACK_NAME`
 
-The controller that maintain the status of the replica-set should be deployed in 1 single instance over a Swarm manager.
+Alternatively, you can use the simple [script](deploy.sh) we created that cover both steps:
+
+* `sh deploy.sh`
+
+Allow some time while images are pulled in the nodes and services are deployed. After a couple of minutes, you can check if all services are up, as usual, running:
+
+```
+$ docker service ls
+ID                  NAME                MODE                REPLICAS            IMAGE                              PORTS
+hmld6tiwr5o0        mongo_mongo         global              0/3                 mongo:3.2                          *:27017->27017/tcp
+uppaix6drfps        mongo_controller    replicated          1/1                 martel/mongo-replica-ctrl:latest   
+```  
+
+You can also check the operations performed by the *controller* reading the logs:
+
+```
+$ docker service logs -f mongo_controller
+mongo_controller.1.7x6ujhg5naw7@swarm-1    | INFO:__main__:Waiting mongo service (and tasks) (mongo_mongo) to start
+mongo_controller.1.7x6ujhg5naw7@swarm-1    | ERROR:__main__:Expired attempts waiting for mongo service (mongo_mongo)
+mongo_controller.1.sv8eztwisitz@swarm-manager    | INFO:__main__:Waiting mongo service (and tasks) (mongo_mongo) to start
+mongo_controller.1.sv8eztwisitz@swarm-manager    | INFO:__main__:Mongo service is up and running
+mongo_controller.1.sv8eztwisitz@swarm-manager    | INFO:__main__:To remove: {'10.0.0.3', '10.0.0.4'}
+mongo_controller.1.sv8eztwisitz@swarm-manager    | INFO:__main__:To add: {'10.0.0.7', '10.0.0.6'}
+mongo_controller.1.sv8eztwisitz@swarm-manager    | INFO:__main__:new replSetReconfig: {'ok': 1.0}
+mongo_controller.1.sv8eztwisitz@swarm-manager    | INFO:__main__:Primary is: 10.0.0.6
+```  
+
+To remove the service:
+
+* `docker stack undeploy STACK_NAME`
+* `docker network rm backend`
+
+You can configure the following environment variables for deploying your stack using the provided [`docker-compose.yml`](docker-compose.yml) file (the variables are used in the controller service, so they ar important, without configuring them, the service won't work correctly):
+
+* `MONGO_VERSION`, the default value is `3.2`
```

Comment:
I'd reference the reader to the .env file to see default values. This duplication will quickly become outdated.

Reply:
there is not .env file... :) the default values are in the compose file.

Reply:
for the time being, I won't add one.